src/components/permissions/hooks/usePermissions.js

```diff
@@ -26,23 +26,37 @@ const usePermissions = (permission: Permission, options = {}) => {
     onAllowed = noop,
     onPrompt = noop,
     onDenied = noop,
+    onGetInstructions = noop,
     requestOnMounted = true,
     promptPopup,
     deniedPopup,
   } = options
 
-  const [showDialog] = useDialog()
+  const [showDialog, hideDialog] = useDialog()
   const mountedState = useMountedState()
   const [allowed, setAllowed] = useState(false)
 
   const PromptPopup = promptPopup || promptPopups[permission]
   const DeniedPopup = deniedPopup || deniedPopups[permission]
 
   const showPopup = useCallback(
-    ({ onDismiss = noop, ...props }) =>
+    ({ onDismiss = noop, Content, ...props }) =>
       showDialog({
         isMinHeight: false,
-        onDismiss,
+        onDismiss: () => {
```

Comment:
incapsulate this in ExplanationDialog 
**it shouldn't be here**. usePermissions just should show dialog and nothing more. 
internal dialog's behavior (e.g. additional links etc) should be defined in the dialog we're showing

Reply:
provide some way (context or passing dialog instatce via prop) to component we're displaying could recopmnfigiure dialog (e.g. remove buttons / change text)